### PR TITLE
cli: add a command for diffing revisions

### PR DESCRIPTION
**Pull request by pgavlin**
These changes add a command, `esc env diff`, for displaying the differences between environment revisions. By default, the output includes both value and definition diffs. Diffs between the value as rendered in different formats (e.g. `shell`, `json`, etc.) can also be displayed.

### CODE REVIEW

**Comment by pgavlin** [comment]
We might want to send the output of this command through the pager (ala `git diff`). Thoughts?

File: cmd/esc/cli/env_diff.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	var format string
+	var showSecrets bool
+	var pathString string
+
+	diff := &envGetCommand{env: env}
+
+	cmd := &cobra.Command{
+		Use:   "diff [<org-name>/]<environment-name>[:<revision-or-tag>] [<revision-or-tag>]",
+		Args:  cobra.RangeArgs(1, 2),
+		Short: "Show changes between revisions.",
```
Comment (cleverguy25) [inline cmd/esc/cli/env_diff.go:28]: Should this say versions since it support revisions and tags

File: cmd/esc/cli/env_get.go [hunk 2]
```diff
@@ -155,7 +153,86 @@ func (get *envGetCommand) showValue(
...
+	var base strings.Builder
+	if err := get.writeValue(ctx, &base, orgName, envName, baseRevisionOrTag, path, format, showSecrets); err != nil {
+		return err
+	}
+	var tip strings.Builder
+	if err := get.writeValue(ctx, &tip, orgName, envName, tipRevisionOrTag, path, format, showSecrets); err != nil {
+		return err
+	}
+
+	baseRef := fmt.Sprintf("%s:%s", envName, baseRevisionOrTag)
```
Comment (cleverguy25) [inline cmd/esc/cli/env_get.go:206]: Did we land on @ instead? Assume that would be a separate PR

**Review by cleverguy25 — APPROVED** [review]
Some nits

File: cmd/esc/cli/env_get.go (hunk 2 above)
Comment (pgavlin) [reply cmd/esc/cli/env_get.go:206]: We did--that'll be a separate PR
